docs/inventory_object.md

```diff
@@ -0,0 +1,187 @@
+# inventory directive in kustomization.yaml
+
```

Comment:
nit: capitalize inventory?
nit: The file is called inventory_object.md. Is inventory an object, directive or field in a YAML file, both?

Reply:
`inventory` is a field in kustomization.yaml. When present, `kustomize build` emits an inventory object.